magento/pwa-studio
# Added local URL support for cypress in docker

## Description

Updates the cypress headless tests script to work with local Magento instances. In the previous solution, we had only worked with fully quantified URLs, like the PR instance URLs because they are publicly deployed on the internet. But the local instances are local to your machine and docker when it runs a container creates its own network which is detached from the host machine network. That's the reason why cypress when it runs on docker works to be able to resolve a local app server to run tests on.

This PR solves the problem by handling local URLs differently.

## Related Issue

PWA-2144

### Verification Stakeholders

@sharkySharks 

## Verification Steps

1. Checkout the branch
2. Run `yarn test:headless <local instance URL>` - All tests should run but you can stop after 1 test success, no need to run all
3. Run `yarn test:headless https://develop.pwa-venia.com` - All tests should run but you can stop after 1 test success, no need to run all

**_Note: To stop testing in between, open another terminal and run `docker ps` to get the ID and `docker kill <ID>` to kill the container._**

#### Test scenario(s) for direct fix/feature

None

## Breaking Changes (if any)

None

## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
